Tests/test_inventory.gd

```diff
@@ -0,0 +1,58 @@
+extends GutTest
+
+func before_each() -> void:
+	InventoryManager.reset_inventory()
+
+func test_add_gold() -> void:
+	InventoryManager.gold_component.add_gold(10)
+	assert_eq(InventoryManager.gold_component.get_gold_amount(), 10)
+
+func test_lose_gold() -> void:
+	InventoryManager.gold_component.lose_gold(10)
+	assert_eq(InventoryManager.gold_component.get_gold_amount(), -10)
+
+func test_lose_torch() -> void:
+	InventoryManager.torch_component.lose_torches(10)
+	assert_eq(InventoryManager.torch_component.get_torch_amount(), -10)
+
+func test_add_torch() -> void:
+	InventoryManager.torch_component.add_torches(10)
+	assert_eq(InventoryManager.torch_component.get_torch_amount(), 10)
+
+func test_add_relic() -> void:
+	var test_relic : Relic = load("res://Items/test_relic.tres")
+	InventoryManager.relic_component.add_relic(test_relic)
+	assert_eq(InventoryManager.relic_component.get_held_relics(), [test_relic])
+
+func test_remove_relic() -> void: 
+	var test_relic : Relic = load("res://Items/test_relic.tres")
+	InventoryManager.relic_component.add_relic(test_relic)
+	InventoryManager.relic_component.add_relic(test_relic)
+	InventoryManager.relic_component.add_relic(test_relic)
+	InventoryManager.relic_component.lose_relic(test_relic)
+	assert_eq(InventoryManager.relic_component.get_held_relics(), [test_relic, test_relic])
```

Comment:
could you make instead make two test relics, add them both, then in one test you lose the first relic, and in the second test you lose the second relic ? that's to ensure stuff gets removed properly not matter the order, because here we don't really see in which order everything is, since everything is the same